zap/src/main/java/org/parosproxy/paros/core/scanner/Scanner.java

```diff
@@ -569,4 +572,20 @@ public int getId() {
     public void setId(int id) {
         this.id = id;
     }
+
+    List<ScanFilter> getScanFilters() {
+        return scanFilters;
+    }
+
+    public void addScanFilter(ScanFilter scanFilter) {
+        if (scanFilter != null) {
+            this.scanFilters.add(scanFilter);
+        }
+    }
+
+    public void setScanFilters(List<ScanFilter> scanFilters) {
```

Comment:
Could be removed now.

Reply:
> > Did all the suggested changes.
> 
> I'm afraid not, `FilterResult` is still not exposing a non-filtered result (there's no need to create an instance each time since they are immutable). (I'd suggest copy the existing `ParseFilter.FilterResult` implementation as it already provides the desired behaviour, the JavaDoc just need minor tweaks.)
 Got your point. Will add a new instance of NON_FILTERED in FilterResult class.

Reply:
> Could be removed now.

i think it is ok to give bulk addition of instances of scanfilter instead of calling add again and again.

Reply:
done.